src/CodeSnippetWidget.js

```diff
@@ -13,22 +13,30 @@ export default class CodeSnippetWidget extends React.Component {
     super(props)
     this.clickHandler = this.clickHandler.bind(this)
     this.state = {
-      active: 0
+      activeTab: 0,
+      active: props.har.method + props.har.url + 0
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.har.url !== this.props.har.url) {
+      this.setState({ active: this.props.har.method + this.props.har.url + this.state.activeTab})
     }
   }
 
   getSnippetKey(snippet) {
     return `${snippet.target}${snippet.client ? `-${snippet.client}` : ""}`
   }
 
-  clickHandler(e) {
-    e.preventDefault()
-    this.setState({ active: e.target.id })
+  clickHandler(index) {
+    this.setState({ active: this.props.har.method + this.props.har.url + index, activeTab: index })
   }
 
   render() {
     const { har } = this.props
 
+    const uniqueId = har.method + har.url
```

Comment:
Can we use a computed property for `har.method + har.url` so it is more clear where this pattern is being used? e.g.:
```js
  get harKey() {
    return this.props.har.method + this.props.har.url
  }
```
Then this variable can be `const harKey = this.harKey`, which is more descriptive of which context the key is unique for.